mongodbatlas/resource_mongodbatlas_project_api_key.go

```diff
@@ -13,6 +13,14 @@ import (
 	matlas "go.mongodb.org/atlas/mongodbatlas"
 )
 
+const (
+	orgRolePrefix     = "ORG_"
+	projectRolePrefix = "GROUP_"
+	orgReadOnlyRole   = "ORG_READ_ONLY"
+)
+
+var orgRoleProvided = false
```

Comment:
is there a risk in having this global variable being used in separate operations? Set during create, and then used during read and delete.

Reply:
no, I don't this is an issue but I renamed the var to make it more specific to the project API key resource

Reply:
updated

Reply:
My only doubt is that this might work during acceptance tests as the test framework maintains one plugin gRPC server for the duration of each test case, while in normal Terraform operations the plugin server starts and stops ([reference](https://developer.hashicorp.com/terraform/plugin/sdkv2/testing/acceptance-tests#how-acceptance-tests-work)).
Approving the PR, I would just try a local build to be sure org roles are supported with no issue.

Reply:
> is there a risk in having this global variable being used in separate operations? Set during create, and then used during read and delete.

I think I misread this comment the first time. The reason why I defined this as a global var was to use it in the create and read operations as I need to know if the user has provided any ORG roles in the config:

The main reason for this is that the CREATE endpoint creates an org API key with `ORG_READ_ONLY` role **if the user did not provide any ORG_ROLES as input**. In this scenario, the READ will return the `ORG_READ_ONLY` role (in addition to the `GROUP_*` roles) even if the user provided only GROUP roles

Reply:
I tested with the following example:
```
resource "mongodbatlas_project_api_key" "api_1" {
  description = "test api_key multii"
  project_id  = "<project_id>"

  project_assignment {
    project_id = "<project_id>"
    role_names = ["ORG_READ_ONLY", "GROUP_READ_ONLY"]
  }
}
```

- `terraform apply` creates the api key successfully.
- then running `terraform plan` outputs incorrect diff showing that "ORG_READ_ONLY" will be added.

I believe the `projectAPIKeyOrgRoleProvided` is not preserving its value during the create and following read operation done in `terraform plan`, so it filters out the org role.

Let me know if you are able to reproduce the same, it is indeed an edge case but might be worth addressing.

Reply:
I was able to reproduce the issue: the reason why it did not "fail" during the plan is that I was using the provider in debug mode. Thanks for the discussion here. I will see if I find a way around

Reply:
Unfortunately, I have not found a way to allow the use of `ORG_READ_ONLY` since it is automatically added when ORG roles are not provided. As a result, I updated the logic to now allow the `ORG_READ_ONLY`  for project assignments and update the documentation to explain why. This should not be a breaking change since customers were not able to use this resource with org roles.